cachito/workers/tasks/utils.py

```diff
@@ -97,3 +105,81 @@ def _assert(
             relpath = fullpath.relative_to(self._root_dir)
             err_msg = err_template.format(relpath=relpath)
             raise ValidationError(f"File check failed for {self._pkg_manager}: {err_msg}")
+
+
+def runs_if_request_in_progress(task_fn):
+    """
+    Decorate a task to make it check request state before proceeding.
+
+    If request state is not "in_progress", the task will log an error and exit.
+
+    :param task task_fn: the task to run if its state is in_progress
+    """
+
+    @functools.wraps(task_fn)
+    def task_with_state_check(*args, **kwargs):
+        """Check request state, proceed to execute task if check succeeds."""
+        request_id = get_function_arg_value("request_id", task_fn, args, kwargs)
+        if not request_id:
+            log.error(
+                "Skipping %s task because the request_id was not found", task_fn.__name__,
```

Comment:
If there is no request_id, I'd say we want to raise an error (probably TypeError or ValueError) here. Using this decorator with a function that has no request_id would be a developer error.